include/dem/pw_contact_force.h

```diff
@@ -101,7 +108,10 @@ class PWContactForce
                                           Tensor<1, dim>,
                                           Tensor<1, dim>> &forces_and_torques,
                          Tensor<1, dim> &                  particle_momentum,
-                         Tensor<1, dim> &                  particle_force)
+                         Tensor<1, dim> &                  particle_force,
+                         Point<dim>                        point_on_boundary,
+                         Point<dim>                        center_mass,
```

Comment:
I think it would be better to pass a reference here than pass a copy. Additionnaly, the center_of_mass (or centroid) would be better off stored as a member of the class because it is not changing...

Reply:
Deal